docs/source/rule-testing.rst

```diff
@@ -22,22 +22,27 @@ To test a new rule, first create a new file under ``test/integration/rules`` nam
 
 .. note:: Multiple tests can be included in one file simply by adding them to the "records" array within the `rule_name_goes_here.json` file.
 
-The ``data`` key can either be a Map or a String.  Normally all json types should be in Map format while others should be String (``csv, kv, syslog``).
-
-The ``description`` key should be a short sentence describing the intent of the test.
-
-The ``trigger`` key indicates whether or not this record should produce an alert or not.  This is used to determine that all of our test records produce the correct amount of alerts.
-
-The ``source`` key is the name of the kinesis stream or s3 bucket.  The source value should match a source provided for this service in ``conf/sources.json``.
+Rule Test Reference
+-------------------
 
-The ``service`` key is the name of the AWS service which sent the alert (kinesis or s3).
+=============      ================================= ========  ===========
+Key                Type                              Required  Description
+-------------      --------------------------------- --------- ----------
+``data``           ``{}`` or ``string``              ``true``  All ``json`` log types should be in Map format while others (``csv, kv, syslog``) should be ``string``
+``description``    ``string``                        ``true``  A short sentence describing the intent of the test
+``trigger``        ``boolean``                       ``true``  Whether or not a record should produce an alert
+``trigger_count``  ``integer``                       ``false`` The amount of alerts that should be generated.  Used for nested data
+``source``         ``string``                        ``true``  The name of the Kinesis Stream or S3 bucket.  This value should match a source provided in ``conf/sources.json``
```

Comment:
"The name of the Kinesis Stream or S3 bucket that this data is landing in." (or something similar). Maybe mention the filename where this can be found